src/App.tsx

```diff
@@ -1,14 +1,74 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
+import { TodoModal } from './components/Todo';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilterBy } from './types/Filter';
+
+const filteredTodos = (todos: Todo[], query: string, filteredBy: FilterBy) => {
+  let preparedTodos = [...todos];
+
+  if (query) {
+    const preparedQuery = query.trim().toLowerCase();
+
+    preparedTodos = preparedTodos.filter(todo => todo.title.toLowerCase().includes(preparedQuery));
+  }
+
+  switch (filteredBy) {
+    case FilterBy.ACTIVE:
+      preparedTodos = preparedTodos.filter(todo => !todo.completed);
+      break;
+
+    case FilterBy.COMPLETED:
+      preparedTodos = preparedTodos.filter(todo => todo.completed);
+      break;
+    case FilterBy.All:
+    default:
+      break;
+  }
+
+  return preparedTodos;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.All);
+  const [isDataLoading, setIsDataLoading] = useState(true);
+  const [isLoadingError, setIsLoadingError] = useState(false);
+
+  const visibleTodos = filteredTodos(todos, query, filterBy);
```

Comment:
better to use `useMemo` hook here
```suggestion
  const visibleTodos = useMemo(
    () => filteredTodos(todos, query, filterBy),
    [todos, query, filterBy],
  );
```